### PR TITLE
Added support for filterBy and hideFilterInput props

### PR DESCRIPTION
**Pull request by jasonblanchard**
Added `filterBy` and `hideFilterInput` props to make it easier to control filtering with an external text input in another component.

Internally, this uses the `filterBy` method, so any changes to how that works will automatically be applied when using the `filterBy` prop, too.

### CODE REVIEW

**Comment by glittershark** [comment]
If it's not too much trouble, I'd like the addition of `filterBy` and `hideFilterInput` to be in two separate commits.

**Comment by glittershark** [comment]
That particular nit aside, :+1: :+1: :+1:

**Comment by glittershark** [comment]
well, plus the tests are failing

**Comment by jasonblanchard** [comment]
@glittershark Sure! No problem, I can do that.

Will update and rebase.

**Comment by jasonblanchard** [comment]
@glittershark It looks like all the tests are passing locally and in the output here https://travis-ci.org/glittershark/reactable/builds/87489176

I'm not sure why it's exiting with a status code of 1... maybe the markdownlint:readme? I'll try to make those lines shorter I guess.

**Comment by glittershark** [comment]
yeah, I lint the markdown to have all lines wrapping to 80 chars.

**Comment by jasonblanchard** [comment]
@glittershark Rebased to 2 commits and all tests/linters passing now.

**Comment by glittershark** [comment]
Looks great! I'll pull this down to do some manual testing later today.

**Comment by jasonblanchard** [comment]
Awesome, thanks.

File: tests/reactable_test.jsx [hunk 1]
```diff
@@ -1640,6 +1640,107 @@ describe('Reactable', function() {
...
+                                  <Reactable.Td column='Description'>bacon</Reactable.Td>
+                                  <Reactable.Td column='Tag'>renewed</Reactable.Td>
+                              </Reactable.Tr>
+                          </Reactable.Table>
+                        </div>
+                      );
+                    }
+                  })
+
+                  this.component = ReactDOM.render(React.createElement(ParentComponent), ReactableTestUtils.testNode());
```
Comment (glittershark) [inline tests/reactable_test.jsx]: can this change to use `<ParentComponent />` rather than `React.createElement`?

Comment (glittershark) [inline tests/reactable_test.jsx]: On second thought, this isn't a merge blocker for me. I'll fix it myself

**Comment by glittershark** [comment]
Thanks for the PR!

**Comment by jasonblanchard** [comment]
:+1:

**Comment by jasonblanchard** [comment]
Thanks for merging in! When do you think we'll have a version bump with these changes?

**Pull request merged**